File: survival-client/src/main/java/net/survival/client/input/KeyboardAdapter.java
package net.survival.client.input;

import net.survival.input.Key;

public abstract class KeyboardAdapter
{
    protected void pressKey(Key key) {
        Keyboard.keyState[key.ordinal()] = true;
    }

    protected void releaseKey(Key key) {
        Keyboard.keyState[key.ordinal()] = false;
    }

    public void nextInputFrame() {
        System.arraycopy(Keyboard.keyState, 0, Keyboard.prevKeyState, 0,
                Keyboard.prevKeyState.length);
    }
}

File: survival-client/src/main/java/net/survival/client/input/KeyboardAdapter.kt
package net.survival.client.input

abstract class KeyboardAdapter {
    protected fun pressKey(key: Key) {
        Keyboard.keyState[key.ordinal] = true
    }

    protected fun releaseKey(key: Key) {
        Keyboard.keyState[key.ordinal] = false
    }

    fun nextInputFrame() {
        System.arraycopy(Keyboard.keyState, 0, Keyboard.prevKeyState, 0, Keyboard.prevKeyState.size)
    }
}